pkg/kube/http_utils.go

```diff
@@ -71,6 +74,38 @@ func GetData(key string, s *corev1.Secret) (string, error) {
 	return auth, nil
 }
 
+// GetDataFromRegistrySecret retrieves the given key from the secret as a string
+func GetDataFromRegistrySecret(key string, s *corev1.Secret) (string, error) {
+	dockerConfigJsonEncoded, ok := s.StringData[key]
```

Comment:
Will this always be a _Docker_ registry? Shouldn't we be as repo-agnostic as possible?
Edit: I see there is a `Type:  kubernetes.io/dockerconfigjson`, so that's this function is meant for.  Are you aware of other registry credentials rather than that one?